src/runner.py

```diff
@@ -158,48 +166,56 @@ def get_info(self):
             )
         return runners
 
-    def reconcile(self, num_runners):
+    def reconcile(self, virt_type, quantity):
         """Bring runners in line with target."""
-        active_runners = []
-        stale_runners = []
-        for runner in self.get_info():
-            if runner.is_active:
-                active_runners.append(runner)
-            else:
-                stale_runners.append(runner)
-
-        if stale_runners:
-            runner_names = ", ".join(r.name for r in stale_runners)
-            logger.info(f"Cleaning up stale runners: {runner_names}")
-            for runner in stale_runners:
+        runners = [r for r in self.get_info() if r.virt_type == virt_type]
+        online_runners = [r for r in runners if r.is_online and r.is_local]
+        offline_runners = [r for r in runners if r.is_offline]
+
+        # Clean up stale (offline) runners
+        if offline_runners:
+            runner_names = ", ".join(r.name for r in offline_runners)
+            logger.info(f"Cleaning up stale {virt_type} runners: {runner_names}")
+            for runner in offline_runners:
                 self._remove_runner(runner)
 
-        delta = num_runners - len(active_runners)
+        # Add/Remove runners to match the target quantity
+        delta = quantity - len(online_runners)
```

Comment:
This answers the question "how many more runners do I need to start locally"
```suggestion
        delta = quantity - len(local_online_runners)
```